sstables/mc/parquet_writer.hh

```diff
@@ -0,0 +1,286 @@
+
+#pragma once
+
+#include <seastar/parquet/parquet/column_writer.h>
+#include <mutation_fragment.hh>
+#include <tombstone.hh>
+#include <dht/i_partitioner.hh>
+#include <range_tombstone.hh>
+#include <seastar/parquet/parquet/file_writer.h>
+#include <seastar/parquet/parquet/schema.h>
+#include <seastar/parquet/parquet/properties.h>
+#include <seastar/core/sstring.hh>
+#include <schema.hh>
+#include <iostream>
+
+namespace parquet {
+
+// TODO will be removed
+std::ofstream _trace = std::ofstream("/tmp/parquet-trace.log", std::ofstream::binary);
+
+using pq_schema = std::shared_ptr<schema::GroupNode>;
+using row_buffer = std::unordered_map<column_id, std::vector<bytes>>;
+using sstable_schema = ::schema;
+
+
+class parquet_writer {
+
+private:
+
+
+    static constexpr auto test_keyspace = "mk";
+
+    const std::string _file_tag;
+    const sstable_schema &_sstable_schema;
+    const pq_schema _pq_schema;
+    std::shared_ptr<seastarized::FileFutureOutputStream> _pq_ostream;
+    std::shared_ptr<seastarized::ParquetFileWriter> _pq_file_writer;
+    row_buffer _buffer;
+
+
+    std::ostream &trace(const std::string &tag) {
+        return _trace << "[" << tag << ":" << _file_tag << "]";
+    }
+
+    Type::type map_column_type(abstract_type::kind sstable_type) {
+        using _type = abstract_type::kind;
+        switch (sstable_type) {
+            case _type::int32:
+                return Type::INT32;
+            case _type::utf8:
+                return Type::BYTE_ARRAY;
+            default:
+                return Type::UNDEFINED; // unsupported type
+        }
+    }
+
+
+    pq_schema convert_to_parquet(const sstable_schema &src_schema) {
+        schema::NodeVector fields;
+
+        const auto &columns = src_schema.all_columns();
+        for (const auto &col_def : columns) {
+            auto name = col_def.name_as_text();
+            auto sstable_type = col_def.type->get_kind();
+            auto parquet_type = map_column_type(sstable_type);
+            if (parquet_type != Type::UNDEFINED)
+                fields.push_back(schema::PrimitiveNode::Make(
+                        name, Repetition::REQUIRED,
+                        parquet_type, ConvertedType::NONE));
+        }
+
+        return std::static_pointer_cast<schema::GroupNode>(
+                schema::GroupNode::Make("schema", Repetition::REQUIRED, fields));
+    }
+
+    std::shared_ptr<seastarized::FileFutureOutputStream>
+    create_pq_ostream() {
+        std::string file = std::string(_file_tag);
```

Comment:
_file_tag is just a tag name used for debugging to differentiate the logs per sstable.